### PR TITLE
Remove RethinkDB

### LINKED ISSUE #39: Remove RethinkDB
It will come back... just as a separate gem!

### PR DESCRIPTION
**Pull request by darrencauthon**
Resolves #39.

Just because all of this code was deleted doesn't mean it's not coming back.  I would like to still keep RethinkDB support... it should just be in a separate gem.

This PR removes all references to rethink.  This was easier than I thought it would be, since RethinkDB was already isolated from the rest of the library.

### CODE REVIEW

**Comment by darrencauthon** [comment]
Sleep on it.

**Comment by timboisvert** [comment]
I'm in favor of this. Not sure I'll go back to rethink. If we want to do it in the future we can pull an old branch, grab the code, and produce a gem. I vote for merging this.

**Comment by darrencauthon** [comment]
Yeah, is rather build the pg hstore and mongo gems before rethink.  

Just for setting expectations - I'll almost always put a PR up for review, and sleep on it before I merge. The review is as much for me as everybody else, as I've found that it's useful to put some distance between myself and the code I just wrote before committing it. If nobody objects and some time passed, I'll merge. 

Like now!

File: lib/kilt/engine.rb [hunk 1]
```diff
@@ -14,4 +13,4 @@ class Engine < ::Rails::Engine
     config.generators.test_framework :rspec
           
   end
-end
\ No newline at end of file
+end
```
Comment (darrencauthon) [inline lib/kilt/engine.rb:16]: This drives me nuts...

**Comment by darrencauthon** [comment]
For the record, I took great care in making sure that the rethink functionality was preserved through all of the refactoring that happened to it. Farewell, RethinkDB.  It was me, not you.

**Pull request merged**